FeatureExtraction/Texture/texture_features.py

```diff
@@ -0,0 +1,412 @@
+"""
+
+import numpy as np
+from copy import copy
+from typing import List, Dict
+from math import pow, log, sqrt, exp, degrees
+from data_types import NumpyImageUINT8, GLCM
+from skimage.feature import greycomatrix, greycoprops
+
+
+def construct_glcm(image: NumpyImageUINT8, distances: List[int],
+                   angles: List[float]) -> GLCM:
+    if not isinstance(image, NumpyImageUINT8):
+        raise TypeError("image should be an instance of NumpyImageUINT8")
+    glcm = greycomatrix(image, distances=distances, angles=angles, normed=True,
+                        levels=256)
+    return GLCM(glcm, distances, angles)
+
+
+def calculate_contrast(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    return greycoprops(glcm, 'contrast')
+
+
+def calculate_dissimilarity(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    return greycoprops(glcm, 'dissimilarity')
+
+
+def calculate_homogeneity(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    return greycoprops(glcm, 'homogeneity')
+
+
+def calculate_correlation(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    return greycoprops(glcm, 'correlation')
+
+
+def calculate_energy(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    return greycoprops(glcm, 'energy')
+
+
+def calculate_autocorrelation(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    y_shape, x_shape = glcm.shape[0:2]
+    distances = glcm.shape[2]
+    angles = glcm.shape[3]
+    results = np.zeros((distances, angles))
+    for distance in range(0, distances):
+        for angle in range(0, angles):
+            x_indexes = np.linspace(1, x_shape, x_shape)
+            y_indexes = np.linspace(1, y_shape, y_shape)
+            xv, yv = np.meshgrid(x_indexes, y_indexes)
+            result = np.sum(
+                np.multiply(glcm[:, :, distance, angle], np.multiply(xv, yv)))
+            results[distance, angle] = result
+    return results
+
+
+def calculate_cluster_prominence(glcm: GLCM):
+    if not isinstance(glcm, GLCM):
+        raise TypeError("glcm should be an instance of GLCM class")
+    y_shape, x_shape = glcm.shape[0:2]
+    distances = glcm.shape[2]
+    angles = glcm.shape[3]
+    results = np.zeros((distances, angles))
+    for distance in range(0, distances):
+        for angle in range(0, angles):
+            result = 0
+            for i in range(0, y_shape):
+                row_mean = np.mean(glcm[i, :])
+                for j in range(0, x_shape):
+                    column_mean = np.mean(glcm[:, j])
+                    result += pow(i + 1 + j + 1 - row_mean - column_mean, 4) * \
```

Comment:
Wouldn't it look nicer if you would call i - row and j - column?

Reply:
Changed here and in all other functions.